rpcserver.go

```diff
@@ -6841,3 +7185,133 @@ func getDecimalDisplayNonStrict(
 
 	return fn.Some(decDisplay), nil
 }
+
+// rfqChannel returns the channel to use for RFQ operations. If a peer public
+// key is specified, the channels are filtered by that peer. If there are
+// multiple channels for the same asset, the user must specify the peer public
+// key.
+func (r *rpcServer) rfqChannel(ctx context.Context, id asset.ID,
+	peerPubKey *route.Vertex) (*channelWithAsset, error) {
+
+	balances, err := r.computeChannelAssetBalance(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error computing available asset "+
+			"channel balance: %w", err)
+	}
+
+	assetBalances, haveBalance := balances[id]
+	if !haveBalance || len(assetBalances) == 0 {
+		return nil, fmt.Errorf("no asset channel balance found for "+
+			"asset %s", id.String())
+	}
+
+	// If a peer public key was specified, we always want to use that to
+	// filter the asset channels.
+	if peerPubKey != nil {
+		assetBalances = fn.Filter(
+			assetBalances, func(c channelWithAsset) bool {
+				return c.channelInfo.PubKeyBytes == *peerPubKey
+			},
+		)
+	}
+
+	switch {
+	// If there are multiple asset channels for the same asset, we need to
+	// ask the user to specify the peer public key. Otherwise, we don't know
+	// who to ask for a quote.
+	case len(assetBalances) > 1 && peerPubKey == nil:
+		return nil, fmt.Errorf("multiple asset channels found for "+
+			"asset %s, please specify the peer pubkey", id.String())
+
+	// If the user specified a peer public key, and we still have multiple
+	// channels, it means we have multiple channels with the same asset and
+	// the same peer. So we just return the first channel.
+	case len(assetBalances) >= 1:
+		return &assetBalances[0], nil
```

Comment:
nit: the first channel may be depleted, need to think of rotation

Reply:
At this level the actual channel balance doesn't matter. We only need the `channel` object to get the peer public key. We don't want to do balance checks at this level really, as that should be handled by the sending logic.